src/main/java/com/oath/halodb/HaloDBInternal.java

```diff
@@ -281,6 +286,66 @@ int get(byte[] key, ByteBuffer buffer) throws IOException {
         }
     }
 
+    public boolean takeSnapShot() {
+
+        logger.info("Start generating the snapshot");
+
+        File[] filesToCopy = dbDirectory.getPath().toFile()
+            .listFiles(file -> {
```

Comment:
Any reason not copying all files including ```META``` and ```LOCK```. When do partition migration, these files are also needed, right?

Reply:
Existing META would introduce misleading information if adding to the snapshot. And Lock is just some runtime objects which not need to be copied.